handbook/engineering/cloud/index.md

```diff
@@ -37,4 +89,4 @@ _Updated 2020-07-03_
 
 We are hiring for these roles:
 
-- +1 [Software Engineer - Backend](https://github.com/sourcegraph/careers/blob/master/job-descriptions/software-engineer-backend.md)
+- +1 Software Engineer - Full Stack (Job req TBD)
```

Comment:
Are we not hiring more backend engineers for this team? I think it might be valuable to change this to a more prose "Growth plan" like other team pages: https://about.sourcegraph.com/handbook/engineering/search#growth-plan, https://about.sourcegraph.com/handbook/engineering/web#growth-plan

Update date on L88.

Reply:
> Are we not hiring more backend engineers for this team? I think it might be valuable to change this to a more prose "Growth plan" like other team pages.

It's worth discussing with the team what growth needs we can identify currently beyond the full stack engineer role. I'll come back and convert this to a growth plan like other teams after that discussion (next cloud sync).